ReactWindows/ReactNative/Modules/Camera/CameraRollManager.cs

```diff
@@ -0,0 +1,156 @@
+                }
+
+                QueryOptions queryOption;
+
+                switch ((string)(options["assetType"]))
+                {
+                    case "All":
+                        queryOption = new QueryOptions(CommonFileQuery.OrderByDate, SupportedVideoFileTypes.Concat(SupportedImageFileTypes).ToList());
+                        break;
+                    case "Videos":
+                        queryOption = new QueryOptions(CommonFileQuery.OrderByDate, SupportedVideoFileTypes);
+                        break;
+                    case "Photos":
+                    default:
+                        queryOption = new QueryOptions(CommonFileQuery.OrderByDate, SupportedImageFileTypes);
+                        break;
+                }
+
+                StorageFolder picturesFolder = KnownFolders.CameraRoll;
+                queryOption.FolderDepth = FolderDepth.Deep;
+
+                IReadOnlyList<StorageFile> sortedItems = await picturesFolder.CreateFileQueryWithOptions(queryOption).GetFilesAsync(0, limit);
+                
+                var result = new JObject();
+                var edges = new JArray();
+                var pageinfo = new JObject();
+                var endcursor = "";
+
+                foreach (StorageFile file in sortedItems)
+                {
+                    var node = new JObject();
+                    var nodeObj = new JObject();
+                    var image = new JObject();
+                    var location = new JObject();
+                    var uri = new Uri(file.Path);
+
+                    ImageProperties imageProperties = await file.Properties.GetImagePropertiesAsync();
+
+                    // Put ImageInfo
+                    image.Add("uri", uri);
+                    image.Add("width", imageProperties.Width);
+                    image.Add("height", imageProperties.Height);
+
+                    // Put BasicNodeInfo
+                    node.Add("image", image);
+                    node.Add("type", file.ContentType);
+                    node.Add("timestamp", imageProperties.DateTaken.ToUnixTimeSeconds());
+                    node.Add("group_name", file.DisplayName);
+
+                    // Put LocationInfo
+                    if (imageProperties.Latitude != null && imageProperties.Longitude != null)
+                    {
+                        location.Add("latitude", imageProperties.Latitude);
+                        location.Add("longitude", imageProperties.Longitude);
+                        node.Add("location", location);
+                    }                    
+                    
+                    nodeObj.Add("node", node);
+                    edges.Add(nodeObj);
+
+                    endcursor = imageProperties.DateTaken.ToString();
+                }
+
+                // Put PageInfo
+                pageinfo.Add("has_next_page", limit < sortedItems.Count);
+                pageinfo.Add("end_cursor", endcursor);
```

Comment:
If you don't implement the `after` filter in your query, I suspect there are cases where this may always return `true`. You'll always grab the first N photos, and if there are more photos, then `first` will always be less than `sortedItems.Count`.